http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala

```diff
@@ -130,10 +130,29 @@ object persistentDiskQuery extends TableQuery(new PersistentDiskTable(_)) {
   def updateSize(id: DiskId, newSize: DiskSize, dateAccessed: Instant) =
     findByIdQuery(id).map(d => (d.size, d.dateAccessed)).update((newSize, dateAccessed))
 
-  def getIsGceFormatted(id: DiskId, dateAccessed: Instant)(implicit ec: ExecutionContext): DBIO[Option[FormattedBy]] =
+  def getFormattedBy(id: DiskId)(implicit ec: ExecutionContext): DBIO[Option[FormattedBy]] =
     findByIdQuery(id).map(_.formattedBy).result.headOption.map(_.flatten)
 
-  // TODO add other queries as needed
+  def isDiskAlreadyAttached(diskId: DiskId)(implicit ec: ExecutionContext): DBIO[Boolean] =
+    for {
+      formattedBy <- getFormattedBy(diskId)
+      r <- formattedBy match {
+        case None =>
+          for {
+            isAttachedToRuntime <- RuntimeConfigQueries.isDiskAttached(diskId)
+            isAttached <- if (isAttachedToRuntime) DBIO.successful(true)
+            else appQuery.isDiskAttached(diskId)
+          } yield isAttached
+        case Some(FormattedBy.Galaxy) =>
+          appQuery.filter(x => x.diskId.isDefined && x.diskId === diskId).length.result.map(_ > 0)
```

Comment:
Wouldn't this return `true` if we had done the following?
- created a disk and attached it to a Galaxy app
- deleted the app but kept the disk

In other words, is it correct to assume that if a disk was formatted by Galaxy, it is attached?

Reply:
No. If we delete app, we "dettach" (meaning deleting the diskId reference from either app table or cluster table)

Reply:
👍